src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -7969,15 +8174,102 @@ class C
 {
     static bool M(ReadOnlySpan<char> chars) => chars is null;
 }";
-            CreateCompilationWithSpanAndMemoryExtensions(source, parseOptions: TestOptions.RegularPreview)
-                .VerifyDiagnostics(
-                    // (5,57): error CS0037: Cannot convert null to 'ReadOnlySpan<char>' because it is a non-nullable value type
+            CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, references: new[] { Net50.SystemMemory }, parseOptions: TestOptions.RegularPreview)
+                .VerifyEmitDiagnostics(
+                    // (5,57): error CS0150: A constant value is expected
```

Comment:
It feels like the previous diagnostic was easier to understand.

Reply:
> It feels like the previous diagnostic was easier to understand.

Agreed. To be clear though, the diagnostic changed because the `ReadOnlySpan<T>` type declaration in NetCoreApp has different conversion operators than the `TestSources` declaration we were using previously.